### PR TITLE
Remove explainBottom

### LINKED ISSUE #2639: Remove explainBottom
The function `explainBottom` is a no-op, and it should be removed. We should replace it with a log entry that captures the match data, after #2396.

### PR DESCRIPTION
**Pull request by andreiburdusa**
Fixes #2639 

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

**Comment by andreiburdusa** [comment]
```
kore> test (suite: kore-test)
                  
kokroer-er-erpelp:l :[ 1[016492]2 ]W aWranrinnign g( W(aWranrUnnUinfiyfByoBtottotmo)m:)
: 
      d idsitsitnicntc ti nitnetgeegresr
s 
      W hWehne nuk nouirnfeiy-firyneigpn:lg
::  
 [  5  5  7 k 2 o ] r   e/W -*a/r r*eFn pliFl nl:Fg  n F[ (n1DW 0 aD4Sr 2fnS]aUf  naWCi alfCr yln*B i/o*n t/g\t  do\(vmdW{)vaI:{rn
Int nU{ tn} {i} }f(d}y"i(B0s"o"t0t)i"t
n)o kc
m ot ) r  : ei 
W-n  irtW teei hpgt :lehd
:r:i  s
s [
 t 1  i 4  n 1  c 3  t ]W    h i/We n*an/t r *eFnu glinFe nilrFgf sn yF
 (in DWn   agD Sr:  fn
SWaU fh n aeCi  nlf C  y lu*B  n/o *i t /f\t/ ydo*\ivm dn{)FvgI:l{:n
 I
t Fn { nt }  { } D} (d } "iS( 1sf" 0ta1 "i 0/)nC"*
cl) t 
F *li/kk n ooFkt\rrnoedee rgv--Dee{rr -rIeeSrsnppfe
tllap {:: l }  C: }[[l  (41 [W"85*5h005/2e"04 1n)3]\7 
] d2u  Wv]n Wa{ i arIWf rnnayWnitriiin{nntng}ighg }n:: ((g

(W"   Wa0(  ar"W  rn)a  nU
r  Un n  ni U  if n  fy i//yBWf**Boiy  ottBFFtthollto:t  om
tFFm) onn): m  :
 )DD
  :     
SS    ff    aa d    di/ CCis*dllst i  tiFs**inlt//nc i  ctFn\\t ncdd i tvvinD {{nt iIIteSnnnegftttgeae{{er g}}rsCe}}s
lr((
  s""  *
01  / "0    )" W\ 
)Whd  
hevW en{h n Ie  unnWunt ini{utif}nhfy}i:yi(f
in"y ng1i g:0n :
"g 
 ):   

      k    o  / r  * e    -  F r/ l/e*  *p /F lF*nF:l  l  FD [Fl F6n Sn5 Ff 9DnaD1    6SDCS]f lf aS aW f* aCa/Crl  ln C\ i*ld*n/ v/g *{  \/I\(d ndWkv\tvao{d{{rrIv}Inen{}nU-tI(tnr{n"{ie}t1}fp}{0}yl(}"(B:"})"o 0(
1t["""t6)1)o4
"
m9 ) )7 
 :0   
]     W W Wi i atWt rhihdn:t:ii
h
sn : tg 
 i    n(   cW   ta    r   in   nU/ /tn* *ei / gfF*Feyl lrB F soFlF
tn n t F  oDnD m    )SkDSW:fo fh
arSae  ef n C-aC  lr lu  eC nd*pl*ii/l /fs :* yt\ /\iid[ dnnv1\vgc{5d{:tI5vI
 n7{n it]It n{ n{ t}Wt} e}a{} g(r}( e"n}" r1i(1 s0n"0/
"g1"* ) 0)  
("
F W)l a
 WrFhnneU kknnkDoo io rrufrSeenyef--iB-arrfor eeyteCppitplllnol ::gm:*  :) /[[
:[ 51 
1\30  0d41  3v]8  5{ 3  9IW0 d4na] i]tr  s {nW/tW}ia*ia}nr nr(gnFcn" ilti1(n  n"WgFig)a nn 
r( t( nWDeW Ua ga nrSer infrnWfUasUiyn 
ntBiC ihofl f:ty  y
tB* B oo/Wo mt ht )t\et :odno 
mv m  ){u)  :In:  
ni
/  tf *d {y  i }i Fs }n ltd(gd ii":iFns1
snct" t ti) iD n
 n ic  cSnt  tft    aei  i gnW nCeti/tlret*e sgh g*
e:Fe/ r
lr  s  s\ 
 F
d   n vW    {h  D Ie    nnW SWt h fh{ue/ae}nn* n}i  C (fuFlu"ynl n1ii *i0nfF/f"gyn y):i \i

nDdn g vg :S{: 
fI
  an    t   C{   l}    } / *( * /"    0 F/\"/l*d)*  v
 FF{ FnlI l  n  DFt F n{WnS }i fD}tDa (h  S":SCf1
fla0 a  "  *C) C/kl
 l o   \r* *de/ /v-   {r\/\Ied*dnpv vtl{F{{:IlI} n n}[tFt(1{n{"1} }08}D}"6( (k)9"S"o
92f2r ]"a"e  ) )- W
C
r a l eWr   pin * lti / :hnW W :gi\i[
 tdt1 (hvh2 W:{:3 a
I
6 r n 5 n t 9 U { ] n }   i } W/f ( a*y " r B 1 nFo/0/ilt*"*n t ) gFoF
F nmll( )  WD:FFa 
nnrS   nf DDUa   n  SSiCdfffliaay s  B*tCCo/illt n  t\c**odt//mv   ){i\\:Indd
ntvv te{{ {gII }enn }rttd(s{{i"
}}s1 }}t0 ((i" ""n) 11c
W00th"" e))in

n tueknkgoioerfrreyes-i-
rnr ege p:p l
l : :W   h[ [e1 1n7 9 9 0u4 1n3 4i6/7f]*]y   iWFWnalagr r:nFn
ini n n gDg     (S( WfW aaa r r nCn/UlU*n n i*iFf/fly y B\BFodontvt t{tDoIo mnmS)t)f:{:a
}
  } C ( l "   2 *d"d/i)i s
s\t tdi ivn n{c cItWtn i titi{nhn}t:t}e
e(g g"e e2r r"s s)
 

             /  W*WWh hieFetnlnh   :uFu
nnn i i fDf y y iSi nfn gag : : 
C
/ l *     * F / l     \ F d n v  /{/D*I*  n SFtFfl{la }  F}FCn(nl "  D1D* 0 /S"S f)f\a
ad  vCkC{lolI r n*e*t/-/{ r }\e\}dpd(vlv"{:{1I I0n[n"t2t){2{
}6}}4}(k0("o1"3r]3"e ")-W)
ra
 er  pn  li  :n W gWi[ it2(th2Wh:5a:
5r
 8n  9U  ]n   i  Wf  ay  rB  no /it/*nt* go F mFl()l W: Fa
Fnr n n  DU D n  SidSffifaysa Bt CoiCltnl tc *ot*/m / )i \:n\d
tdv ev{ g{I eIn rntdst{i
{}s }}t }(i ("n "1cW10th0" e")in)
n 
tueknogireferky-soir
rne egp -:l r
: e  Wp [hl 2e: 6n  2 [ 5u2 7n8 8i1/]f8* y3 iW8Fna]lgr  :nWF
ian nr  gnD  i  (nS Wgf a a r(  nWC/Ual*nr  in*FfU/lyn  Bi\Fofdntyv tB{DooI mtnS)ttf:o{a
m}  )}C :(l 
"   3*d "/i ) s 
\td dii vns {ct ItiWn nitict{nth}t :}ei
(gn "et 3re "sg )
e 
 r   s   
    / W *Wh  ie FtnWlh h :ueF
nnn i   fuD yn  iiS nff gya :i  
nC/ gl* :   
*F  /l      \F  dn  v / {D* I   nSF/tfl*{a  } FF}Cnl(l  " DF1* n0/S " fD)\a 
d SvCf{laI  n*Ct/l{  }\*}d/(v "{\1Id0nv"t{){I
}n}tk({o"}r4}e"(-)"r
4e "p )l 
:   W [i 3t 9hW7:i5
t7 h2 :] 
 k  Wo  ar  re  n-  ir/ ne* gp   lF/(:l*W   a[FFr3nln9  U1DFn3 ni8S f4fDy]a B  SoWCftalatr  on*Cmi/l)n  :g\*
 d/ (v  W{\ aId rnvdnt{iU{Isn}nti}tif({ny"}cB1}to0( t""it)1no
0tm"e)k)g:o
e
rr es -
 r  e dp il s:Wt hi[en4nc1 t6u 5ni0in4ft]ye igWneagrr:sn

i  n  g     ( WW ha er nn/ U*un niFiflfy yBFionnt gtD:o 
mS )f :a 
   C  l     * d//i *s\ tdFivln{ cIFtnn t i{Dn} t}Se(fg"ae4 r"Cs)l

   *  /     \WWdhivet{nhI :nu
tn {i }f }y (i "n 4g ": )
/
 *     F  l   W Fi nt  h/D:* 
 S Ff la    FC nl    D*  //S *f\ adF vlC{ lIF nn*t /{D } \}Sd(fv"a{1 I0Cn"lt) {
*}/} (\"d5v"{)I
n t { } }W(i"t1h0:"
) 
  k o r e - r/e*p lF:l  [F4n4 6D7 3S4f]a  WCalr n*i/n g\ d(vW{aIrnntU{n}i}f(y"B1o0")
kotrteo-mr)e:p
l :   [ 4d8i3s2t5i3n]c tW airnntiengge r(sW
a r n U Wnhiefny Buontitfoymi)n:g
: 
      d i s t i n/c*t  Filn tFeng eDr sS
f a   C lW h*e/n  \udnvi{fIynitn{g}:}
( " 5 " ) 
      / *W iFtlh :F
n   D   S f a   C/l*  *F/l  \Fdnv {DI nStf{a} }C(l" 6*"/) 
\ d v { IWnitt{h}:}
( " 1 0 " ) 
  /* Fl Fn D Sfa Cl */ \dv{Int{}}("10")
kore-repl: [521926] Warning (WarnUnifyBottom):
    distinct integers
    When unifying:
        /* Fl Fn D Sfa Cl */ \dv{Int{}}("6")
    With:
        /* Fl Fn D Sfa Cl */ \dv{Int{}}("10")
kore-repl: [538622] Warning (WarnUnifyBottom):
    distinct integers
    When unifying:
        /* Fl Fn D Sfa Cl */ \dv{Int{}}("7")
    With:
        /* Fl Fn D Sfa Cl */ \dv{Int{}}("10")
kore-repl: [527969] Warning (WarnUnifyBottom):
    distinct integers
    When unifying:
        /* Fl Fn D Sfa Cl */ \dkvo{rIen-tr{e}p}l(:" 7["5)4
9 457] Warning  ( W aWrintUhn:i
f y B o t t o m )/:*
  F l   Fdni sDt iSnfcat  Ciln t*e/g e\rdsv
{ I n t {W}h}e(n" 1u0n"i)f
ying:
        /* Fl Fn D Sfa Cl */ \dv{Int{}}("8")
    With:
        /* Fl Fn D Sfa Cl */ \dv{Int{}}("10")
kore-repl: [588407] Warning (WakronrUen-irfeypBlo:t t[o5m7)5:6
3 8 ]   Wdairsntiinngc t( WianrtneUgneirfsy
B o t t oWmh)e:n
  u n i fdyiisntgi:n
c t   i n t e g e/r*s 
F l   F nW hDe nS fuan iCfly i*n/g :\
d v { I n t { } }/(*" 9F"l) 
F n   D  WSiftah :C
l   * /   \ d v {/I*n tF{l} }F(n" 8D" )S
f a   C lW i*t/h :\
d v { I n t { } }/(*" 1F0l" )F
n D Sfa Cl */ \dv{Int{}}("10")
kore-repl: [592852] Warning (WarnUnifyBottom):
    distinct integers
    When unifying:
        /* Fl Fn D Sfa Cl */ \dv{Int{}}("9")
    With:
        /* Fl Fn D Sfa Cl */ \dv{Int{}}("10")
TEST RUN PASSED

Duration:  30.663s
Passed:    4829
Failed:    0


kore> Test suite kore-test passed
Completed 2 action(s).
```

File: kore/src/Kore/Log/DebugUnifyBottom.hs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+debugUnifyBottom info first second =
+    logEntry $
+        DebugUnifyBottom
+            info
+            (TermLike.mapVariables (pure $ from @_ @VariableName) first)
+            (TermLike.mapVariables (pure $ from @_ @VariableName) second)
+
+debugUnifyBottomAndReturnBottom ::
+    MonadLog log =>
+    Alternative log =>
```
Comment (ana-pantilie) [inline kore/src/Kore/Log/DebugUnifyBottom.hs:85]: Shouldn't this be `MonadUnify` instead?

File: kore/src/Kore/Repl/Data.hs [hunk 2]
```diff
@@ -688,31 +621,21 @@ makeKoreReplOutput :: String -> ReplOutput
...
-runUnifierWithExplanation ::
+runUnifierWithoutExplanation ::
     forall m a.
     MonadSimplify m =>
-    UnifierWithExplanation m a ->
-    m (Either ReplOutput (NonEmpty a))
-runUnifierWithExplanation (UnifierWithExplanation unifier) =
-    failWithExplanation <$> unificationResults
+    UnifierT m a ->
+    m (Maybe (NonEmpty a))
```
Comment (ana-pantilie) [inline kore/src/Kore/Repl/Data.hs:628]: Why was this changed to a `Maybe`?

Comment (andreiburdusa) [reply kore/src/Kore/Repl/Data.hs:628]: I thought there is no need for `ReplOutput` because there would be no `explainBottom`. The explanations would be printed as log entries instead of being wrapped in a `Left`

Comment (ana-pantilie) [reply kore/src/Kore/Repl/Data.hs:628]: What happens if the result is `Nothing`? Do we just not print anything anymore?

Comment (andreiburdusa) [reply kore/src/Kore/Repl/Data.hs:628]: Yes, [here](https://github.com/kframework/kore/pull/2674/files#diff-b87f2e448b6098a71a0953983f8941bae7a8d3f93aa3284c54ce2cb1d7a3efc5R1619). I don't know any means to tell if the log entry was called or not in order to tell the user "No explanation given" in the appropriate case.

Comment (ana-pantilie) [reply kore/src/Kore/Repl/Data.hs:628]: Ok, I understand now. Thanks for explaining.

**Review by ana-pantilie — APPROVED** [review]

**Review by emarzion — APPROVED** [review]
